ERDMaker/Services/DiagramBuilder.cs

```diff
@@ -2,37 +2,57 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Text;
 
 namespace ERDMaker.Services
 {
-    public class DiagramBuilder
+    public static class DiagramBuilder
     {
-        public static string StringifyTable(Table table)
+        internal static string StringifySchema(IEnumerable<SchemaDefinition> definitions)
         {
-            var references = new List<string>();
+            var definitionStrings = definitions.Select(StringifyElement);
+            return string.Join(Environment.NewLine, definitionStrings);
+        }
 
-            var lines = new List<string>
+        private static string StringifyElement(SchemaDefinition definition)
+        {
+            switch (definition)
             {
-                $"Table {table.Name} {{"
-            };
-            foreach (var field in table.Fields)
+                case OptionSet p:
+                    return StringifyOptionSet(p);
+                case Table t:
+                    return StringifyTable(t);
+            }
+
+            throw new ArgumentOutOfRangeException(definition.GetType().Name);
+        }
+
+        private static string StringifyOptionSet(OptionSet optionSet)
+        {
+            var optionSetBuilder = new StringBuilder();
+
+            optionSetBuilder.AppendLine($"Enum {optionSet.Name}_enum {{");
+            foreach (var value in optionSet.Values)
             {
-                lines.Add($"\t{field.Name} {field.Type} {field.Decoration}");
-                foreach (var reference in field.References)
-                {
-                    references.Add($"Ref: {table.Name}.{field.Name} > {reference}");
-                }
+                optionSetBuilder.AppendLine($"\t\"{value}\"");
```

Comment:
there is a bug when the optionset label contains quotation mark

Reply:
There is no way to escape the quotation mark in the DBML spec.  We'll have to settle for a simple replacement to a single quote